examples/voice-assistant/assistant_calls.py

```diff
@@ -0,0 +1,117 @@
+    LIVING_ROOM = "living room"
+    KITCHEN = "kitchen"
+    BATHROOM = "bathroom"
+    OFFICE = "office"
+
+
+class AssistantFnc(llm.FunctionContext):
+    @llm.ai_callable(desc="Turn on/off the lights in a room")
+    async def toggle_light(
+        self,
+        room: Annotated[Room, llm.TypeInfo(desc="The specific room")],
+        status: bool,
+    ):
+        logging.info("toggle_light %s %s", room, status)
+        ctx = AssistantContext.get_current()
+        key = "enabled_rooms" if status else "disabled_rooms"
+        li = ctx.get_metadata(key, [])
+        li.append(room)
+        ctx.store_metadata(key, li)
+
+    @llm.ai_callable(desc="User want the assistant to stop/pause speaking")
+    def stop_speaking(self):
+        pass  # do nothing
+
+
+async def entrypoint(ctx: JobContext):
+    gpt = openai.LLM(model="gpt-4-turbo")
+
+    assistant = VoiceAssistant(
+        vad=silero.VAD(),
+        stt=deepgram.STT(),
+        llm=gpt,
+        tts=elevenlabs.TTS(),
+        fnc_ctx=AssistantFnc(),
+    )
+
+    @assistant.on("will_synthesize_llm")
+    def _will_synthesize_llm(chat_ctx: llm.ChatContext, user_msg: llm.ChatMessage):
+        # inject system prompt
+        chat_ctx.messages.insert(
+            0,
+            llm.ChatMessage(
+                role=llm.ChatRole.SYSTEM,
+                text="You are a voice assistant created by LiveKit. Your interface with users will be voice. You should use short and concise responses, and avoiding usage of unpronouncable punctuation.",
+            ),
+        )
+
+    @assistant.on("agent_speech_interrupted")
+    def _agent_speech_interrupted(chat_ctx: llm.ChatContext, msg: llm.ChatMessage):
+        msg.text += "... (user interrupted you)"
+
+    @assistant.on("function_calls_done")
+    def _function_calls_done(ctx: AssistantContext):
+        logging.info("function_calls_done %s", ctx)
+        enabled_rooms = ctx.get_metadata("enabled_rooms", [])
+        disabled_rooms = ctx.get_metadata("disabled_rooms", [])
+
+        async def _handle_answer():
+            prompt = "Make a summary of the following actions you did:"
+            if enabled_rooms:
+                enabled_rooms_str = ", ".join(enabled_rooms)
+                prompt += f"\n - You enabled the lights in the following rooms: {enabled_rooms_str}"
+
+            if disabled_rooms:
+                disabled_rooms_str = ", ".join(disabled_rooms)
+                prompt += f"\n - You disabled the lights in the following rooms {disabled_rooms_str}"
+
+            chat_ctx = llm.ChatContext(
+                messages=[llm.ChatMessage(role=llm.ChatRole.SYSTEM, text=prompt)]
+            )
+
+            stream = await gpt.chat(chat_ctx)
+            await assistant.say(stream)
+
+        if enabled_rooms or disabled_rooms:
+            asyncio.ensure_future(_handle_answer())
+
+    # start the assistant with the first participant found inside the room
+
+    @ctx.room.on("participant_connected")
```

Comment:
i wonder if it's better to have `assistant.start` handle this? i.e. if you don't pass in a participant, it'll automatically wait for the first participant. wdyt?

Reply:
Just wondering the case where there is multiple people in the room. but I guess it is OK as a default impl